rust/psibase/src/services/tokens/mod.rs

```diff
@@ -100,17 +93,17 @@ mod service {
     }
 
     #[action]
-    fn credit(tokenId: TID, receiver: AccountNumber, amount: Quantity, memo: String) {
+    fn credit(tokenId: TID, receiver: AccountNumber, amount: Quantity, memo: Memo) {
```

Comment:
This doesn't work as well as it does in C++. In C++, the `Memo` type will fail to unpack if the constraint is violated (>80 bytes). In rust, it only works if you construct a `Memo` with `new()`, which won't happen when unpacking.

Therefore, you should impl `validate()` on `Memo`, and every action that takes a memo needs to also call `memo.validate()`

Reply:
@James-Mart I thought by making `contents` a private field, there would be no other option in constructing it unless it was with `Self::new` , is that not the case?

Reply:
Should it circumvent construction do I wanna panic or return a `Result`?